src/java.base/share/classes/sun/nio/cs/CharsetMapping.java

```diff
@@ -244,7 +239,7 @@ void readSINGLEBYTE() {
     void readINDEXC2B() {
         char[] map = readCharArray();
         for (int i = map.length - 1; i >= 0; i--) {
-            if (c2b == null && map[i] != -1) {
+            if (c2b == null) {
```

Comment:
Post-review comment:

I don't quite understand this code. `c2b` is loop-invariant, right? So, what this does is checking for `c2b != null`, and if so, instantiates the `c2b` array with `new char[map[map.length - 1] + 256]`. This does not even require a loop, as written right now.

Reply:
Indeed the code looks weird. Will have a deeper look. Thanks!

Reply:
The code seems to read the data written with this piece of code:
https://github.com/openjdk/jdk/blob/b55e418a077791b39992042411cde97f68dc39fe/make/jdk/src/classes/build/tools/charsetmapping/JIS0213.java#L145
but apparently it does not seem to work as expected. Since this is a very old code, not sure I would proactively modify it though.